yb-voyager/cmd/assessMigrationCommand.go

```diff
@@ -63,7 +65,21 @@ var assessMigrationCmd = &cobra.Command{
 		validateSourceSchema()
 		validatePortRange()
 		validateSSLMode()
-		validateAssessmentMetadataDirFlag()
+		if cmd.Flags().Changed("assessment-metadata-dir") {
+			validateAssessmentMetadataDirFlag()
+			cmd.Flags().VisitAll(func(f *pflag.Flag) {
+				if f.Changed {
+					if strings.Contains(strings.ToLower(f.Name), "source") && f.Name != "source-db-type" {
+						utils.ErrExit("No need of `--source-*` connection related flags as `--assessment-metadata-dir` is provided.\nPlease re-run the command without these flags")
```

Comment:
Let's phrase it as "cannot pass source-* flags when `-assessment-metadata-dir` is passed" ?